consensus/types/src/light_client_bootstrap.rs

```diff
@@ -1,46 +1,63 @@
 use super::{BeaconState, EthSpec, FixedVector, Hash256, SyncCommittee};
 use crate::{
-    light_client_update::*, test_utils::TestRandom, ForkName, ForkVersionDeserialize,
-    LightClientHeader,
+    light_client_header::{
+        LightClientHeaderAltair, LightClientHeaderCapella, LightClientHeaderDeneb,
+    },
+    light_client_update::*,
+    ChainSpec, ForkName, ForkVersionDeserialize, LightClientHeader, SignedBeaconBlock,
 };
 use serde::{Deserialize, Deserializer, Serialize};
 use serde_json::Value;
 use ssz_derive::{Decode, Encode};
 use std::sync::Arc;
-use test_random_derive::TestRandom;
 
 /// A LightClientBootstrap is the initializer we send over to lightclient nodes
 /// that are trying to generate their basic storage when booting up.
-#[derive(
-    Debug,
-    Clone,
-    PartialEq,
-    Serialize,
-    Deserialize,
-    Encode,
-    Decode,
-    TestRandom,
-    arbitrary::Arbitrary,
-)]
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Encode, Decode, arbitrary::Arbitrary)]
 #[serde(bound = "T: EthSpec")]
 #[arbitrary(bound = "T: EthSpec")]
 pub struct LightClientBootstrap<T: EthSpec> {
     /// The requested beacon block header.
-    pub header: LightClientHeader,
+    pub header: LightClientHeader<T>,
     /// The `SyncCommittee` used in the requested period.
     pub current_sync_committee: Arc<SyncCommittee<T>>,
     /// Merkle proof for sync committee
     pub current_sync_committee_branch: FixedVector<Hash256, CurrentSyncCommitteeProofLen>,
 }
 
 impl<T: EthSpec> LightClientBootstrap<T> {
-    pub fn from_beacon_state(beacon_state: &mut BeaconState<T>) -> Result<Self, Error> {
+    pub fn from_beacon_state(
+        chain_spec: &ChainSpec,
+        beacon_state: &mut BeaconState<T>,
+        block: SignedBeaconBlock<T>,
+    ) -> Result<Self, Error> {
```

Comment:
It is unlikely that we will ever create bootstrap messages from the BeaconState in practice. Similar to my comment above, the existence of these functions come from a literal implementation of the spec, but may not survive latter revisions. We can merge this code for now, but let's keep it mind that it may be deleted soon.

Reply:
I'm curious if you have any optimisation approach for this in mind? if not, I'll just create an issue and add it to the [tracking issue](https://github.com/sigp/lighthouse/issues/3651) to make sure that we don't miss this before turning light client server on.